src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1530,6 +1531,19 @@ public static Map<PackType, Integer> readTypedPackFormats(JsonObject json)
         return map.buildOrThrow();
     }
 
+    public static void writeTypedPackFormats(JsonObject json, PackMetadataSection section)
+    {
+        int packFormat = section.getPackFormat();
+        for (PackType packType : PackType.values())
+        {
+            int format = section.getPackFormat(packType);
+            if (format != packFormat)
+            {
+                json.addProperty("forge:" + packType.bridgeType.name().toLowerCase(Locale.ROOT) + "_pack_format", format);
```

Comment:
For the sake of removing redundancy, the creation of the property key (`"forge:" + ... + "_pack_format"`) should be moved to a private helper method near these two methods.